restapi/admin_users.go

```diff
@@ -467,3 +475,65 @@ func getAddUsersListToGroupsResponse(session *models.Principal, params admin_api
 
 	return nil
 }
+
+func getListUsersWithAccessToBucketResponse(session *models.Principal, bucket string) ([]string, *models.Error) {
+	ctx := context.Background()
+	mAdmin, err := newMAdminClient(session)
+	if err != nil {
+		return nil, prepareError(err)
+	}
+	// create a minioClient interface implementation
+	// defining the client to be used
+	adminClient := adminClient{client: mAdmin}
+
+	users, err := listUsers(ctx, adminClient)
+	if err != nil {
+		return nil, prepareError(err)
+	}
+	var retval []string
+	seen := make(map[string]bool)
+	for i := 0; i < len(users); i++ {
+		policy, err := adminClient.getPolicy(ctx, users[i].Policy)
+		if err == nil {
+			parsedPolicy, err2 := parsePolicy(users[i].Policy, policy)
+			if err2 == nil && policyMatchesBucket(parsedPolicy, bucket) {
+				retval = append(retval, users[i].AccessKey)
+				seen[users[i].AccessKey] = true
+			}
+			if err2 != nil {
+				console.Println(err2)
+			}
+		} else {
+			console.Println(err)
+		}
+	}
+
+	groups, err := listGroups(ctx, adminClient)
+	if err != nil {
+		console.Println(err)
```

Comment:
```suggestion
		return nil, prepareError(err) 
```

Here if you dont have groups you should stop processing the request and just return an error, otherwise `len(*groups)` will still crash the operation